### PR TITLE
Azure Monitor Exporter: Add new Integration test projects

### PR DESCRIPTION
**Pull request by TimothyMothra**
This PR introduces Integration Test projects (Test & WebApp).
This PR is limited to just adding the new projects and confirming that the build system passes.

These include basic tests to validate that the project's structure is working. 
More tests will be added at a later date to fully validate Telemetry objects.

>   Integration tests evaluate an app's components on a broader level than unit tests. 
>   Unit tests are used to test isolated software components, such as individual class methods. 
>   Integration tests confirm that two or more app components work together to produce an expected result, 
>   possibly including every component required to fully process a request.
>   https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-3.1

### CODE REVIEW

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <!-- 
+  Integration tests evaluate an app's components on a broader level than unit tests. 
+  Unit tests are used to test isolated software components, such as individual class methods. 
+  Integration tests confirm that two or more app components work together to produce an expected result, 
+  possibly including every component required to fully process a request.
+  https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-3.1
```
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj:8]: nit: sanitize the URL (e.g. removing `en-us` and the `?view=aspnetcore-3.1` parts).

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <!-- 
+  Integration tests evaluate an app's components on a broader level than unit tests.
```
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj:4]: curious: do we need this?

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  <!-- 
+  Integration tests evaluate an app's components on a broader level than unit tests. 
+  Unit tests are used to test isolated software components, such as individual class methods. 
+  Integration tests confirm that two or more app components work together to produce an expected result, 
+  possibly including every component required to fully process a request.
+  https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-3.1
+  -->
+  
+  <PropertyGroup>
+    <RequiredTargetFrameworks>netcoreapp2.1</RequiredTargetFrameworks>
```
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj:12]: Is this temporary, or the intention is test against multiple versions of runtime?

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests/xunit.runner.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+  "maxParallelThreads": 1,
+  "parallelizeTestCollections": false
+}
```
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests/xunit.runner.json:4]: nit: add a blank line before EOF to make GitHub happier.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.csproj [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.Testing" VersionOverride="2.1.1" />
+
+    <PackageReference Include="OpenTelemetry.Instrumentation.AspNetCore" VersionOverride="0.6.0-beta.1" />
+    <PackageReference Include="OpenTelemetry.Extensions.Hosting" VersionOverride="0.6.0-beta.1" />
+    <PackageReference Include="OpenTelemetry.Instrumentation.Http" VersionOverride="0.6.0-beta.1" />
+
+    <PackageReference Include="xunit" />
+    <PackageReference Include="xunit.runner.visualstudio" />
```
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.csproj:32]: curious: why would some packages versioned while this one not?

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.csproj [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    <NoWarn>
+      NETSDK1005; <!-- NETSDK1005: Ensure that restore has run and that you have included 'net461' in the TargetFrameworks for your project. -->
+    </NoWarn>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.Testing" VersionOverride="2.1.1" />
+
+    <PackageReference Include="OpenTelemetry.Instrumentation.AspNetCore" VersionOverride="0.6.0-beta.1" />
```
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.csproj:27]: Consider extracting these versions to a common file before it becomes unmanageable.
Can be a separate PR so no need to get blocked here.

**Review by reyang — APPROVED** [review]
LGTM with some minor suggestions.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj (hunk 2 above)
Comment (TimothyMothra) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj:4]: Are you asking about the comment or the project?

I included the comment because this is an uncommon project type.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj (hunk 3 above)
Comment (TimothyMothra) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj:12]: temporary-ish.
The current frameworks include 4.6.1, which is not supported by the `Microsoft.AspNetCore.Mvc.Testing` framework.
This is also the reason why the pipeline is failing. I've reached out to the Azure SDK folks to get guidance on how to correctly handle this.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj (hunk 2 above)
Comment (reyang) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj:4]: The comment - it seems to be out of scope for this project (e.g. we don't try to define/clarify what integration test is in this project).

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.csproj [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  <!-- https://docs.microsoft.com/aspnet/core/test/integration-tests -->
+  
+  <PropertyGroup>
+    <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.Testing" VersionOverride="2.1.1" />
+
+    <PackageReference Include="OpenTelemetry.Instrumentation.AspNetCore" VersionOverride="0.6.0-beta.1" />
```
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.csproj:11]: nit, sort alphabetically

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.csproj [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    <PackageReference Include="OpenTelemetry.Extensions.Hosting" VersionOverride="0.6.0-beta.1" />
```
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.csproj:12]: Curious, why `VersionOverride` instead of `Version`?

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj [hunk 9]
```diff
@@ -0,0 +1,22 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+  <!-- https://docs.microsoft.com/aspnet/core/test/integration-tests -->
+  
+  <PropertyGroup>
+    <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup Condition=" '$(TargetFramework)' != 'net461' ">
+    <PackageReference Include="Microsoft.AspNetCore.App" />
```
Comment (cijothomas) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj:9]: I think this packagereference is *only* required in netcoreapp2.1. Anything higher than that gets this automatically.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.csproj (hunk 8 above)
Comment (TimothyMothra) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.csproj:12]: Whenever there's a conflict with the packagereferences brought it by Azure SDK, using `VersionOverride` always fixes it.
I defaulted to using it while just trying to get everything to pass in the build pipeline. 

I'll do one more check now and see which can and cannot be removed

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/OpenTelemetry.Exporter.AzureMonitor.sln [hunk 10]
```diff
@@ -11,6 +11,10 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Exporter.Azur
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Exporter.AzureMonitor.Tests", "tests\OpenTelemetry.Exporter.AzureMonitor.UnitTest\OpenTelemetry.Exporter.AzureMonitor.Tests.csproj", "{14546496-6144-49A5-9431-09611F38D30E}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Exporter.AzureMonitor.Integration.Tests", "tests\OpenTelemetry.Exporter.AzureMonitor.IntegrationTests\OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.csproj", "{526C6440-36E1-4385-BC1E-98CD3CB163A3}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp", "tests\OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp\OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj", "{E00C15D6-96E5-4A3B-8701-AC24292301CE}"
```
Comment (reyang) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/OpenTelemetry.Exporter.AzureMonitor.sln:16]: Consider `OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.AspNetCoreWebApp`.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj (hunk 9 above)
Comment (TimothyMothra) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj:9]: ohhhhh good catch! thanks

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj (hunk 3 above)
Comment (TimothyMothra) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests.WebApp/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.WebApp.csproj]: resolved

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.csproj (hunk 7 above)
Comment (TimothyMothra) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.csproj:11]: done

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/OpenTelemetry.Exporter.AzureMonitor.sln [hunk 11]
```diff
@@ -11,6 +11,10 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Exporter.Azur
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Exporter.AzureMonitor.Tests", "tests\OpenTelemetry.Exporter.AzureMonitor.UnitTest\OpenTelemetry.Exporter.AzureMonitor.Tests.csproj", "{14546496-6144-49A5-9431-09611F38D30E}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Exporter.AzureMonitor.Integration.Tests", "tests\OpenTelemetry.Exporter.AzureMonitor.IntegrationTests\OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.csproj", "{526C6440-36E1-4385-BC1E-98CD3CB163A3}"
```
[the remark this answers is not in the document]
Comment (TimothyMothra) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/OpenTelemetry.Exporter.AzureMonitor.sln:14]: THIS project needs to end in the word ".Tests" for the Azure SDK build system.
This project will have access to the Exporter's internals and needs to be public signed. 

It's safe to rename the other project

[the remark this answers is not in the document]
Comment (reyang) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/OpenTelemetry.Exporter.AzureMonitor.sln]: The name sounds like a test project for "all the integration test cases".
Consider something concrete (e.g. if this is a common test lib, put "Common"; if this is a console test project, put "Console").

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/OpenTelemetry.Exporter.AzureMonitor.csproj [hunk 12]
```diff
@@ -9,7 +9,7 @@
 
   <ItemGroup>
     <PackageReference Include="System.Text.Json" />
-    <PackageReference Include="OpenTelemetry" Version="0.6.0-beta.1" IsImplicitlyDefined="true" />
```
Comment (TimothyMothra) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/OpenTelemetry.Exporter.AzureMonitor.csproj:12]: @rajkumar-rangaraj I moved the version to the Packages.Data.props.
I don't tell if the `IsImplicitlyDefined` is required.

This is the only doc page I could find: https://docs.microsoft.com/en-us/nuget/consume-packages/install-use-packages-visual-studio#update-a-package

Please advise.

File: eng/Packages.Data.props [hunk 13]
```diff
@@ -87,6 +87,10 @@
     <PackageReference Update="NSubstitute" Version="3.1.0" />
     <PackageReference Update="nunit" Version="3.12.0" />
     <PackageReference Update="NUnit3TestAdapter" Version="3.13.0" />
+    <PackageReference Update="OpenTelemetry" Version="0.6.0-beta.1" />
+    <PackageReference Update="OpenTelemetry.Extensions.Hosting" Version="0.6.0-beta.1" />
+    <PackageReference Update="OpenTelemetry.Instrumentation.AspNetCore" Version="0.6.0-beta.1" />
+    <PackageReference Update="OpenTelemetry.Instrumentation.Http" Version="0.6.0-beta.1" />
```
Comment (TimothyMothra) [inline eng/Packages.Data.props:93]: @pakrym This is the change we discussed. Please confirm that this is appropriate.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.csproj (hunk 5 above)
Comment (TimothyMothra) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.IntegrationTests/OpenTelemetry.Exporter.AzureMonitor.Integration.Tests.csproj]: In most cases, the version number is supplied by Azure SDK's build properties. 
Now that I have everything working, I'm discussing with Pavel about which are appropriate to move to the common properties

File: eng/Packages.Data.props (hunk 13 above)
Comment (pakrym) [reply eng/Packages.Data.props:93]: LGTM!

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/OpenTelemetry.Exporter.AzureMonitor.csproj (hunk 12 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/OpenTelemetry.Exporter.AzureMonitor.csproj:12]: CI has failed without `IsImplicitlyDefined` hence added it. Without this attribute if CI could pass then we don't need to worry about it.

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/OpenTelemetry.Exporter.AzureMonitor.sln (hunk 11 above)
[the remark this answers is not in the document]
Comment (TimothyMothra) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/OpenTelemetry.Exporter.AzureMonitor.sln]: Yes, I had intended this project to be the test runner for all integration test cases.

I've renamed this to "OpenTelemetry.Exporter.AzureMonitor.Integration.Console.Tests".

**Review by rajkumar-rangaraj — APPROVED** [review]
